### PR TITLE
Adding a new keyboard, the [n]2

### PR DESCRIPTION
**Pull request by NightlyBoards**
## Description

N2 is a 2-key macropad with an integrated atmega32u4 chip.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nightly_boards/n2/config.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0x0717
+#define PRODUCT_ID 0x0003
+#define DEVICE_VER 0x0001
+#define MANUFACTURER Neil Brian Ramirez
+#define PRODUCT [n]2
+#define DESCRIPTION A 2-key pcb
```
Comment (fauxpark) [inline keyboards/nightly_boards/n2/config.h:25]:
```suggestion
```

File: keyboards/nightly_boards/n2/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_L,   
+		KC_O
+    ),
+
```
Comment (fauxpark) [inline keyboards/nightly_boards/n2/keymaps/default/keymap.c:23]:
```suggestion
        KC_L,
        KC_O
    )
```

File: keyboards/nightly_boards/n2/readme.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+![n2](https://i.imgur.com/iHOxmHi.png)
+
+A 2 key pcb
+
+* Keyboard Maintainer: [Neil Brian Ramirez](https://github.com/NightlyBoards)
+* Hardware Supported: atmega32u4
+
+Make example for this keyboard (after setting up your build environment):
+
+    make nightly_boards/n2:default
```
Comment (fauxpark) [inline keyboards/nightly_boards/n2/readme.md:12]:
```suggestion
    make nightly_boards/n2:default
    
Flashing example for this keyboard:

    make nightly_boards/n2:default:flash
```

File: keyboards/nightly_boards/n2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/nightly_boards/n2/rules.mk:11]:
```suggestion
```

File: keyboards/nightly_boards/n2/rules.mk [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/nightly_boards/n2/rules.mk:31]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/nightly_boards/n2/readme.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Alter
+
+![n2](https://i.imgur.com/iHOxmHi.png)
```
Comment (noroadsleft) [inline keyboards/nightly_boards/n2/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![n2](https://i.imgur.com/iHOxmHil.png)
```

Comment (NightlyBoards) [reply keyboards/nightly_boards/n2/readme.md]: https://github.com/qmk/qmk_firmware/pull/9862/commits/b53013853d923d898dee317a66ed8c05eea79334

File: keyboards/nightly_boards/n2/rules.mk (hunk 5 above)
Comment (NightlyBoards) [reply keyboards/nightly_boards/n2/rules.mk]: https://github.com/qmk/qmk_firmware/pull/9862/commits/b694b9859f620daa570d7ccf118270ada0089a0c

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**